sdl_android/src/main/java/com/smartdevicelink/encoder/SdlEncoder.java

```diff
@@ -187,11 +187,15 @@ public void drainEncoder(boolean endOfStream) {
 				}
 
 				if (mBufferInfo.size != 0) {
+					ByteBuffer encoderOutputBuffer = encoderOutputBuffers[encoderStatus];
 					byte[] dataToWrite = new byte[mBufferInfo.size];
-					encoderOutputBuffers[encoderStatus].get(dataToWrite,
-							mBufferInfo.offset, mBufferInfo.size);
```

Comment:
mBufferInfo.offset indicates the offset of valid data in the codec output buffer, not the destination buffer.
FYI, I tested with Nexus 6 and Xperia Z3+ and this offset value was always 0 on both phones. So it's possible that this value isn't used, or is used only by specific phones.

Please also refer to the "famous" example http://bigflake.com/mediacodec/EncodeAndMuxTest.java.txt.